piglow/examples/strobe.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/goiot/devices/piglow"
+	"golang.org/x/exp/io/i2c"
+)
+
+func main() {
+	p, err := piglow.Open(&i2c.Devfs{Dev: "/dev/i2c-1", Addr: piglow.Address})
+	if err != nil {
+		panic(err)
+	}
+
+	// catch signals and terminate the app
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc,
+		syscall.SIGHUP,
+		syscall.SIGINT,
+		syscall.SIGTERM,
+		syscall.SIGQUIT)
+
+	if err := p.Setup(); err != nil {
+		panic(err)
+	}
+
+	for {
+		select {
+		case <-sigc:
+			p.Shutdown()
+			p.Close()
```

Comment:
Sounds a little bit like an exaggeration, doesn't it ;) I disagree with `impossible to read` and in regards to `look for the fundamental reference parts`, I think that if that's what you're looking for as a user, you should look at godoc. On the other hand, I think having real examples of how people should be writing real code is extremely valuable. GoBot suffers from having many examples that aren't realistic and as soon as you need to figure something like handling a clean exit, things get hard for newbies.